lib/apiservers/engine/backends/system.go

```diff
@@ -375,6 +375,12 @@ func (s *System) AuthenticateToRegistry(ctx context.Context, authConfig *types.A
 		token, err := fetcher.FetchAuthToken(authURL)
 		if err != nil || token.Token == "" {
 			log.Errorf("Fetch auth token failed: %s", err)
+			// At this point, if a request cannot be solved by a retry, it is an authentication error.
+			switch err.(type) {
```

Comment:
Nit: since we're only concerned with the `urlfetcher.DoNotRetry` type here, an `if` condition will suffice:

```go
if _, ok := err.(urlfetcher.DoNotRetry); ok {
    err = fmt.Errorf("Get %s: unauthorized: incorrect username or password", loginURL)
    return "", err
}
```

Switch is fine too - feel free to keep it that way if you'd like.